lib/commands/performance.js

```diff
@@ -104,91 +181,44 @@ helpers.getBatteryInfo = async function getBatteryInfo (dataReadTimeout = 2) {
-  return await retryInterval(dataReadTimeout, RETRY_PAUSE, async () => {
-    let cmd = ['dumpsys', 'meminfo', `'${packageName}'`, '|', 'grep', '-E', "'Native|Dalvik|EGL|GL|TOTAL'"];
-    let data = await this.adb.shell(cmd);
-    if (!data) throw new Error('No data from dumpsys'); //eslint-disable-line curly
-
-    let totalPrivateDirty, totalPss,
-        nativePrivateDirty, nativePss, nativeHeapSize, nativeHeapAllocatedSize,
-        dalvikPrivateDirty, dalvikPss,
-        eglPrivateDirty, eglPss,
-        glPrivateDirty, glPss;
-    let apilevel = await this.adb.getApiLevel();
-    for (let line of data.split('\n')) {
-      let entries = line.trim().split(' ').filter(Boolean);
-      // entries will have the values
-      //   ['<System Type>', '<Memory Type>', <pss total>, <private dirty>, <private clean>, <swapPss dirty>, <heap size>, <heap alloc>, <heap free>]
-      // except 'TOTAL', which skips the second type name
-      //
-      // and on API level 18 and below
-      //   ['<System Type', '<pps>', '<shared dirty>', '<private dirty>', '<heap size>', '<heap alloc>', '<heap free>']
-
-      if (apilevel > 18) {
-        let type = entries[0];
-        let subType = entries[1];
-        if (type === 'Native' && subType === 'Heap') {
-          // native heap
-          nativePss = entries[2];
-          nativePrivateDirty = entries[3];
-          nativeHeapSize = entries[6];
-          nativeHeapAllocatedSize = entries[7];
-        } else if (type === 'Dalvik' && subType === 'Heap') {
-          // dalvik heap
-          dalvikPss = entries[2];
-          dalvikPrivateDirty = entries[3];
-        } else if (type === 'EGL' && subType === 'mtrack') {
-          // egl
-          eglPss = entries[2];
-          eglPrivateDirty = entries[3];
-        } else if (type === 'GL' && subType === 'mtrack') {
-          // gl
-          glPss = entries[2];
-          glPrivateDirty = entries[3];
-        } else if (type === 'TOTAL' && entries.length === 8) {
-          // there are two totals, and we only want the full listing, which has 8 entries
-          totalPss = entries[1];
-          totalPrivateDirty = entries[2];
-        }
+helpers.getMemoryInfo = async function getMemoryInfo (packageName, retries = 2) {
+  return await retryInterval(retries, RETRY_PAUSE_MS, async () => {
+    const cmd = [
+      'dumpsys', 'meminfo', `'${packageName}'`,
+      '|', 'grep', '-E',
+      `'${MEMINFO_TITLES.NATIVE}|${MEMINFO_TITLES.DALVIK}|${MEMINFO_TITLES.EGL}` +
+      `|${MEMINFO_TITLES.GL}|${MEMINFO_TITLES.TOTAL}'`
+    ];
+    const data = await this.adb.shell(cmd);
+    if (!data) {
+      throw new Error('No data from dumpsys');
+    }
+    const valDict = {totalPrivateDirty: ''};
+    const apiLevel = await this.adb.getApiLevel();
+    for (const line of data.split('\n')) {
+      const entries = line.trim().split(/\s+/).filter(Boolean);
+      if (apiLevel >= 30) {
```

Comment:
purely aesthetically would it make more sense to say `> 29` since that is what the function is named?